src/ga4gh/vrs/utils/hgvs_tools.py

```diff
@@ -1,26 +1,333 @@
+import re
 import hgvs
+import hgvs.parser
+import hgvs.dataproviders.uta
+import hgvs.normalizer
+import hgvs.variantmapper
+import hgvs.sequencevariant
 
+import logging
+
+from ga4gh.vrs.dataproxy import _DataProxy, create_dataproxy
+from ga4gh.vrs import models
+
+_logger = logging.getLogger(__name__)
 
 class HgvsTools():
+    """
+    A convenience class that exposes only the tools needed by vrs-python for working with HGVS (Human Genome Variation Society) notation.
 
-    """ A utility class that provides access to hgvs components that have a UTA database dependency"""
-    def __init__(self, uta_conn):
+    Attributes:
+        parser (hgvs.parser.Parser): The HGVS parser object.
+        uta_conn: The UTA (Universal Transcript Archive) connection object.
+        normalizer: The HGVS normalizer object.
+        variantMapper: The HGVS variant mapper object.
+    """
+    hgvs_re = re.compile(r"[^:]+:[cgmnpr]\.")
 
-        self.uta_conn = uta_conn
-        self.normalizer = hgvs.normalizer.Normalizer(self.uta_conn)
+    def __init__(self,dp: _DataProxy = None):
         self.parser = hgvs.parser.Parser()
+        self.uta_conn = None
+        self.normalizer = None
+        self.variantMapper = None
+        self.dp = dp
+
+    def _fetch_uta_conn(self):
+        # just in time instantiation of the UTA connection
+        if self.uta_conn is None:
+            self.uta_conn = hgvs.dataproviders.uta.connect()
+        return self.uta_conn
+    
+    def _fetch_normalizer(self):
+        # just in time instantiation of the normalizer service, with validation set to True
+        if self.normalizer is None:
+            self.normalizer = hgvs.normalizer.Normalizer(self._fetch_uta_conn(), validate=True)
+        return self.normalizer
+    
+    def _fetch_variantMapper(self):
+        # just in time instantiation of the variantMapper service
+        if self.variantMapper is None:
+            self.variantMapper = hgvs.variantmapper.VariantMapper(self._fetch_uta_conn())
+        return self.variantMapper
```

Comment:
IMO, I think we should just initialize these in the `__init__` method

Reply:
+1